File: app/src/main/java/com/gdi/hahi/mvp/presenter/NeihanPresenter.java
package com.gdi.hahi.mvp.presenter;

import android.content.Context;

import com.gdi.hahi.mvp.contract.HahiContract;
import com.gdi.hahi.mvp.model.NeihanModel;
import com.gdi.hahi.mvp.model.bean.Neihan;
import com.gdi.hahi.network.Api;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.functions.Consumer;
import io.reactivex.schedulers.Schedulers;

/**
 * Created by gandi on 2017/8/24 0024.
 */

public class NeihanPresenter implements HahiContract.Presenter {

    NeihanModel hotModel;
    HahiContract.View mView;
    Context context;

    public NeihanPresenter(HahiContract.View mView, Context context) {

        this.mView = mView;
        this.context = context;
    }

    public void setContentType(String content_type) {
        hotModel = new NeihanModel(content_type, Api.device_id);
    }


    @Override
    public void requestData(long lastTime) {
        Observable<Neihan> observable = hotModel.loadData(context, lastTime);
        observable.subscribeOn(Schedulers.io())
                .unsubscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(new Consumer<Neihan>() {
                    @Override
                    public void accept(Neihan neihan) throws Exception {
                        mView.setData(hotModel.convertToDoMain(neihan));
                    }
                });
    }
}

File: app/src/main/java/com/gdi/hahi/mvp/presenter/NeihanPresenter.kt
package com.gdi.hahi.mvp.presenter

import android.content.Context

import com.gdi.hahi.mvp.contract.HahiContract
import com.gdi.hahi.mvp.model.NeihanModel
import com.gdi.hahi.mvp.model.bean.NeiHanData
import com.gdi.hahi.network.Api

import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import kotlin.properties.Delegates

/**
 * Created by gandi on 2017/8/24 0024.
 */

class NeihanPresenter(internal var mView: HahiContract.View<List<NeiHanData>>, internal var context: Context) : HahiContract.Presenter {

    internal var hotModel: NeihanModel by Delegates.notNull()

    fun setContentType(content_type: String) {
        hotModel = NeihanModel(content_type, Api.device_id)
    }


    override fun requestData(lastTime: Long) {
        val observable = hotModel.loadData(context, lastTime)
        observable.subscribeOn(Schedulers.io())
                .unsubscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe { neihan -> mView.setData(hotModel.convertToDoMain(neihan)) }
    }
}
